.github/workflows/graalvm-check.yml

```diff
@@ -0,0 +1,56 @@
+name: GraalVM Check
+
+on:
+  schedule:
+    - cron: '30 18 * * *'
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set up GraalVM
+        uses: graalvm/setup-graalvm@v1
+        with:
+          version: 'latest'
+          java-version: '11'
+          components: 'native-image'
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Check GraalVM installation
+        run: |
+          echo "GRAALVM_HOME: ${{ env.GRAALVM_HOME }}"
+          echo "JAVA_HOME: ${{ env.JAVA_HOME }}"
+          native-image --version
+
+      - name: Grant execute permission for gradlew
+        run: chmod +x gradlew
+
+      - name: Build with Gradle
+        env:
+          packageUser: ${{ secrets.BALLERINA_BOT_USERNAME }}
+          packagePAT: ${{ secrets.BALLERINA_BOT_TOKEN }}
+          JAVA_OPTS: -DBALLERINA_DEV_COMPILE_BALLERINA_ORG=true
+        run: |
+          ./gradlew build
+
+      - name: Set Up Ballerina
+        uses: ballerina-platform/setup-ballerina@v1
+        with:
+          version: 2201.6.0
+
+      - name: Update dependency versions
+        run: ./gradlew :jaeger-extension-ballerina:updateTomlVerions
+        env:
+          packageUser: ${{ secrets.BALLERINA_BOT_USERNAME }}
+          packagePAT: ${{ secrets.BALLERINA_BOT_TOKEN }}
+          JAVA_OPTS: -DBALLERINA_DEV_COMPILE_BALLERINA_ORG=true
+
+      - name: Run Ballerina tests using the native executable
+        run: bal test --native ./prometheus-extension-ballerina
+
+      - name: Run Ballerina build using the native executable
+        run: bal build --native ./prometheus-extension-ballerina
```

Comment:
Shall we swap the order of build and test steps, to make it an incremental flow?

Reply:
sure